src/main/kotlin/com/dto/ReservationManager.kt

```diff
@@ -0,0 +1,61 @@
+package com.dto
+
+import com.excpetion.ReservationAlreadyStartedException
+import com.excpetion.ReservationNotFoundException
+import com.model.ReservationDao
+import com.parkingrate.ParkingRateEvaluator
+import com.util.ReservationCalculator
+import org.springframework.stereotype.Service
+import java.math.BigDecimal
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.util.Optional
+import javax.transaction.Transactional
+
+
+@Service
+class ReservationManager(private val reservationDao: ReservationDao,
+                         private val reservationConverter: ReservationConverter,
+                         private val reservationCalculator: ReservationCalculator) {
+
+    @Transactional
+    fun createReservation(reservationRequest: ReservationRequest, startTime: LocalDateTime): StartedReservationDto {
+
+        Optional.ofNullable(reservationDao.findStartedReservation(reservationRequest.carLicensePlate))
+                .ifPresent { throw ReservationAlreadyStartedException(reservationRequest.carLicensePlate) }
+
+        return reservationConverter.convertToModel(reservationRequest, startTime)
+                .also { reservationDao.save(it) }
+                .let { reservationConverter.convertToReservationStartedDto(it) }
+    }
+
+    @Transactional
+    fun stopReservation(reservationId: Long, endTime: LocalDateTime): StoppedReservationDto {
+        val reservation = reservationDao.findById(reservationId)
+
+        return reservation
+                .map {
+                    it.endTime = endTime
```

Comment:
czemu mutujesz a nie tworzysz nowy obiekt?

Reply:
zgodnie z zasadą, PUT służy do edytowania istniejącego już zasobu a nie tworzenia nowego.

Reply:
można by tak zrobić aczkolwiek zwiększyłaby się też liczba operacji na bazie, ponieważ musielibyśmy usuwać poprzednią encję i tworzyć nową.

Reply:
dodatkowo id encji by się zmieniało co może być mało intuicyjne, tworzymy rezerwację z id 1, następnie po zakończeniu rezerwacji id się zmienia.

Reply:
jeśli chodzi o całą encję to komentarz zostawiłem -> https://github.com/MrK191/Parking/pull/1/files#r217883705